src/utils/diagrammer.ts

```diff
@@ -347,47 +361,55 @@ export function getDiagramDict(exploreFields: ILookmlModelExploreFieldset, joins
     build[0] && arrangeTables(build[0], 0)
   }
 
-  diagramDict.yOrderLookup = yOrder
+  // Drop any tables that were collected but
+  // lack fields and joins 
+  Object.keys(diagramDict.tableData).forEach(key => diagramDict.tableData[key] === undefined && delete diagramDict.tableData[key])
```

Comment:
Appreciate the comments, they _help_ but this file in particular is pretty over-dense. consider multiple files, consider docstrings for all top level functions, and consider breaking some of these functions up into smaller functions that only do one thing.